index.js

```diff
@@ -109,24 +117,29 @@ const connect = async () => {
             });
         });
 
+        await producer.connect();
+
         return producer;
     } catch (err) {
-        logger.error('kafka broker connection failiure', err);
+        logger.error('kafka broker connection failure', err);
         throw err;
     }
 };
 
 /**
- * sends a message to a kafka topic
+ * sends a message to a kafka topic,
+ * call the connect method to get producer obj
+ * @param {Object} producer the producer object
  * @param {Object} data the data object
  * @param {String} topic the topic name
  */
-const sendMessage = async (data, topic) => {
+const sendMessage = async (producer, data, topic) => {
```

Comment:
Not related to the changes in this PR. Documentation of this method indicates that this generic to send any message to any topic. But the logic looks for specific structure in the data.
Is the producer service meant to be generic? If not, we should update the doc to reflect the same and also rename the method name to something more specific.

Reply:
It's not supposed to be generic, it is for sending Screwdriver to build messages only. We don't have any other use cases for now, plus it would be Screwdriver specific always, but yes I can update it later to be more generic

Reply:
Similar correction need to made in `createMessage`. Need not be addressed in this PR. May be we could move it out of producer-service if it makes sense.